src/OrchardCore.Modules/OrchardCore.Flows/Drivers/FlowPartDisplay.cs

```diff
@@ -14,24 +14,23 @@ namespace OrchardCore.Flows.Drivers
 {
     public class FlowPartDisplay : ContentPartDisplayDriver<FlowPart>
     {
-        private readonly IContentDefinitionManager _contentDefinitionManager;
         private readonly IContentManager _contentManager;
         private readonly IServiceProvider _serviceProvider;
 
         public FlowPartDisplay(
             IContentManager contentManager,
-            IContentDefinitionManager contentDefinitionManager,
             IServiceProvider serviceProvider
             )
         {
-            _contentDefinitionManager = contentDefinitionManager;
             _contentManager = contentManager;
             _serviceProvider = serviceProvider;
         }
 
         public override IDisplayResult Display(FlowPart flowPart, BuildPartDisplayContext context)
         {
-            return Shape<FlowPartViewModel>("FlowPart", m =>
+            var hasItems = flowPart.Widgets.Any();
+
+            return Shape<FlowPartViewModel>(hasItems ? "FlowPart" : "FlowPart_Empty", m =>
```

Comment:
Shouldn't it be an alternate instead of switching the shape type? The alternate can be added directly in the lambda.

Reply:
I don't think an alternate would allow you to hide the shape if the Flow or Bag was empty.

The example in #1305 gives a use case for this (conditionally showing content zones)